accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -1708,28 +1709,36 @@ To orchestrate this, type inference on patterns proceeds in three phases:
 
 3.  **Calculate the static type of the pattern.** Using that value type, recurse
     through the pattern again downwards to the leaf subpatterns filling in any
-    missing types in the pattern. This process may also insert implicit
-    coercions and casts from `dynamic` when values flow into a pattern during
-    matching.
+    missing types in the pattern. This process may also insert casts from
+    `dynamic` when values flow into a pattern during matching.
 
     *For example:*
 
     ```dart
-    T id<T>(T t) => t;
-    (T Function<T>(T), dynamic) record = (id, 'str');
-    var (int Function(int) f, String s) = record;
+    (dynamic, dynamic) record = (123, 'str');
+    var (int n, String s) = record;
     ```
 
     *Since the right-hand is not a record literal, we can't use the pattern's
     context type schema to insert coercions when the record is being created.
-    However, the matched value type `(T Function<T>(T), dynamic)` is allowed by
-    the record pattern's required type `(Object?, Object?)`, the field matched
-    value type `T Function<T>(T)` is allowed by the field required type `int
-    Function(int)`, and the field matched value type `dynamic` is allowed by the
-    field required type `String)`. So the declaration is valid. Coercions are
-    inserted after destructuring each record field before passing them to the
-    field subpatterns. At runtime, when the record is destructured during
-    matching, the coercions are applied. This is specified below.*
+    However, the matched value type `(dynamic, dynamic)` is allowed by the
+    record pattern's required type `(Object?, Object?)`, so the declaration is
```

Comment:
We might also mention the type checking of each record field, in order to establish all the parts needed for the conclusion 'so the declaration is valid':
```suggestion
    record pattern's required type `(Object?, Object?)`, and the matched
    value type `dynamic` for each field is allowed by the required types
    of the fields, i.e., `int` and `String`, so the declaration is
```